### PR TITLE
2483 build unauthorised page with build info

### LINKED ISSUE #2483: Build unauthorized page with build information
image

### PR DESCRIPTION
**Pull request by solazio**
## Done
- Build unauthorised page on `/{snap_name}/builds`

## Issue / Card

Fixes #2483 

## QA

- Pull the branch
- Run the site using the command `./run`
- View the site locally in your web browser at: http://0.0.0.0:8004/{snap_name}/builds
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Compare to [design](https://github.com/canonical-web-and-design/snapcraft.io/issues/2483)
- Click `Log in` button and check the content of the page was changed
- To revoke GitHub token after you finished go to https://github.com/settings/applications

## Desktop
image

## Mobile
image

### CODE REVIEW

**Comment by Lukewh** [comment]
Seems to still show even though I have builds. Also removes the connected repo:
Screenshot_2020-02-04  Builds for Olaris

**Comment by solazio** [comment]
@Lukewh I missed that. I don't have any builds and wasn't able to see that. Will fix it.

**Comment by Lukewh** [comment]
@danielmutis what do you think about something like this for mobile?
Screenshot_2020-02-04  Builds for Lukotron testing

Currently it does this:
Screenshot_2020-02-04  Builds for Lukotron testing (1)

**Comment by danielmutis** [comment]
@Lukewh I think this way of breaking up the illustration works well on mobile. I imagine that the "login to your github" message would go at the top.

@solazio on desktop I the text should be centred and with a narrower width for columns of text so they can align better to the 3 parts of the illustration. Similar to how it's implemented here: https://snapcraft.io/build

**Comment by Lukewh** [comment]
> @Lukewh I think this way of breaking up the illustration works well on mobile. I imagine that the "login to your github" message would go at the top.

Yeah the login to your github was because I was scrolled down and it was "sticky" to the top of the visible area (which wasn't the top of the page). We have sticky headers on all the other pages, so I'd think it makes sense here too?

@danielmutis should the button text be "Login to GitHub" without the logo as on Zeplin?

**Comment by danielmutis** [comment]
@Lukewh

>Yeah the login to your github was because I was scrolled down and it was "sticky" to the top of the visible area (which wasn't the top of the page). We have sticky headers on all the other pages, so I'd think it makes sense here too?

Yes it should be sticky at the top. Alles klar.

>@danielmutis should the button text be "Login to GitHub" without the logo as on Zeplin?

I quite like the one with the logo. I didn't put it in the wires because I thought a UI might look at it after, but I think we should keep the button with the logo.

**Comment by anasereijo** [comment]
@solazio as per your request see below the images saved down separately:

https://assets.ubuntu.com/v1/82de77b0-build.snapcraft.hero-01.svg
https://assets.ubuntu.com/v1/275ddbf9-build.snapcraft.hero-02.svg
https://assets.ubuntu.com/v1/e708bab9-build.snapcraft.hero-03.svg

File: templates/publisher/builds.html [hunk 1]
```diff
@@ -1,58 +1,73 @@
...
+            Your snap is linked to:&ensp;
+            <a href="https://github.com/{{ github_repository }}"
+            class="p-link--external">
+              {{ github_repository }}
+            </a>
+          </p>
+        </div>
+      </div>
+    </div>
+          {% else %}
```
Comment (Lukewh) [inline templates/publisher/builds.html]: This else looks like it's in the wrong place?

File: templates/publisher/builds.html [hunk 2]
```diff
@@ -1,58 +1,73 @@
...
+              {{ github_repository }}
+            </a>
+          </p>
+        </div>
+      </div>
+    </div>
+          {% else %} 
+            {% if not github_repository %}
+              {% if not github_user %}
+            Login to your GitHub account to start building.
```
Comment (Lukewh) [inline templates/publisher/builds.html]: Any reason these are less indented than the ifs?

File: templates/publisher/builds.html [hunk 3]
```diff
@@ -1,58 +1,73 @@
...
-                <span class="p-button__text">Save</span>
-              </button>
+            <a href="https://github.com/{{ github_user['login'] }}"
+              class="p-link--soft">
+              <img src="{{ github_user['avatar_url'] }}"
+                alt="@{{ github_user['login'] }}" class="p-build__avatar">
+              {{ github_user["login"] }}
+            </a>
+            &mid;
+            <a href="#">Change account</a>
```
Comment (Lukewh) [inline templates/publisher/builds.html]: Get rid of this for now - we're not implementing it

File: templates/publisher/builds.html [hunk 4]
```diff
@@ -1,58 +1,73 @@
...
           </div>
         </div>
       </div>
     </div>
-
     <div class="row">
       <hr class="u-no-margin--bottom" />
     </div>
-    {% endif %}
+            {% endif %}
```
Comment (Lukewh) [inline templates/publisher/builds.html]: some crazy indentation in this file

File: webapp/authentication.py [hunk 5]
```diff
@@ -39,6 +39,7 @@ def empty_session(session):
     session.pop("macaroon_discharge", None)
     session.pop("openid", None)
     session.pop("user_shared_snaps", None)
+    session.pop("github_auth_secret", None)
```
Comment (Lukewh) [inline webapp/authentication.py]: should this be in here? Is @jkfran doing it in another branch? If it's already done rebase - as adding this line seems to lower code coverage

**Pull request merged**